taqueria-vscode-extension/src/lib/gui/SandboxesDataProvider.ts

```diff
@@ -0,0 +1,72 @@
+import * as vscode from 'vscode';
+import { OutputLevels, VsCodeHelper } from '../helpers';
+import * as Util from '../pure';
+
+export class SandboxesDataProvider implements vscode.TreeDataProvider<SandboxTreeItem> {
+	constructor(
+		private workspaceRoot: string,
+		private helper: VsCodeHelper,
+	) {}
+
+	getTreeItem(element: SandboxTreeItem): vscode.TreeItem {
+		return element;
+	}
+
+	async getChildren(element?: SandboxTreeItem): Promise<SandboxTreeItem[]> {
+		let pathToDir: Util.PathToDir | null;
+		let config: Util.TaqifiedDir | null;
+		if (!this.workspaceRoot) {
+			pathToDir = null;
+			config = null;
+		} else {
+			try {
+				pathToDir = await Util.makeDir(this.workspaceRoot, this.helper.i18n);
+				config = await Util.TaqifiedDir.create(pathToDir, this.helper.i18n);
+			} catch (e: any) {
+				config = null;
+				await this.helper.notifyAndLogError('Error while loading config, sandboxes list will fail to populate', e);
```

Comment:
At some point we should refactor this to use i18n but this is fine for now!

Reply:
Agree. I should keep i18n in mind for next refactors.